### ISSUE TITLE
Discussion of input temperature parameter to FAO56 equation

### ISSUE THREAD

**Issue by karwosts** [jeroenterheerdt/HAsmartirrigation#70]
As I've been running this, I see that my daily evapotranspiration is quite a lot higher than my reference eto for the month for my zip code. Rainmaster gives me historically 0.18 for July, but for the last two weeks or so I've been calculating eto values of 0.3-0.35 every day. 

I tried digging into your methodology a bit to make sure something is not being miscalculated, since that's a significantly higher value than I was expecting and it would require a costly amount of water to make up that difference. 

As I was reading the 1998 paper, and looking at their examples for calculating daily ETo values, I noticed a discrepancy with your implementation and I was wondering if you had an explanation for it. 

If you follow example 18 here: http://www.fao.org/3/x0490e/x0490e08.htm#penman%20monteith%20equation

You will see that in this example of daily calculation the required envirommental temperature data is only Tmax and Tmin. They then take the mean of these two values to get the Tmean value that gets plugged into [equation 6](http://www.fao.org/3/x0490e/x0490e1o.gif) for T (this is the value "t" passed to fao56_penman_monteith function). 

In your implementation, you do not calculate t as (Tmin+Tmax)/2, but rather take "temp_day" from openweathermap API. I cannot find in their documentation exactly what this value represents, they only define it as "day temperature", but it is quite different than the average of max and min. For example, I just did a request to their onecall api and for the daily value for today they gave me: 

[temp][max] = 29.76
[temp][min] = 15.16
[temp][day] = 27.36

If I calculated t for fao56_penman_monteith as done in the example from the 1998 reference paper, I would have averaged tmax and tmin and got 22.46. You pass the day temp directly as 27.36, which is nearly 5C higher. I don't know if that contiributes to the much higher than expected ETo values I've been seeing. 

What do you think? Do you have any more information about what the "day temperature" is, or why that was chosen instead of the average of max and min as in the fao.org example above?

Thank you very much for reading.

**Comment by jeroenterheerdt** [comment]
I honestly did not read all the papers in detail. The component calls https://github.com/woodcrafty/PyETo/blob/master/pyeto/fao.py#L282 and passes is temp day temperature as t. I can definitely change it to pass in an average instead. I will make that change and make another beta release.

**Comment by jeroenterheerdt** [comment]
@karwosts please install this version to see if this improves the situation: https://github.com/jeroenterheerdt/HAsmartirrigation/releases/tag/v0.0.71beta

**Comment by karwosts** [comment]
Thanks for the quick response. I will caution though I'm not an expert in ETo and the extent of my knowledge comes from just reading about for an hour or two, so please don't take my word as gospel in how it's supposed to be, I could be totally wrong about the value that's supposed to be used. 

I can try this and I'm sure the result will be different, but it's hard for me to say if its more or less accurate. 

Just wanted to get a second opinion and make sure that looks right to you (or anyone else that knows about this stuff).

**Comment by jeroenterheerdt** [comment]
I am not an expert either. Reading about it makes me think it actually makes sense what you said, so let's try it :)

**Comment by karwosts** [comment]
Wow dropped the ETo by 33%. This is a lot closer to my reference value of 0.18 inches  (.22 inches instead of .34 inches)

v0.70:
```
evapotranspiration - day_of_year: 217
t_day: 29.37
t_min: 15.17
t_max: 29.61
t_dew: 13.58
elevation: 87
latitude: 45.625223919538946
RH_hr: 38
u_2: 4.540063026269421
pressure: 1014
coastal: False
calculate_solar_radiation: True
estimate_solrad_from_temp: True
solrad: None
**fao56: 8.727896155430043**
```


v0.71:
```
2021-08-05 16:11:53 INFO (MainThread) [custom_components.smart_irrigation.sensor] 
evapotranspiration - day_of_year: 217
t_day: 29.5
t_min: 15.17
t_max: 29.97
t_dew: 13.7
elevation: 87
latitude: 45.625223919538946
RH_hr: 38
u_2: 4.540063026269421
pressure: 1014
coastal: False
calculate_solar_radiation: True
estimate_solrad_from_temp: True
solrad: None
**fao56: 5.8527521086954595**
```

**Comment by jeroenterheerdt** [comment]
ok if we monitor it over the weekend and then make a new release (non-beta) next week?

**Comment by karwosts** [comment]
Sounds fine to me. Or we can wait a month and see if all our grass dies 😄

**Comment by jeroenterheerdt** [comment]
ok to close this?

**Comment by ScratMan** [comment]
Hello, I'm new to HASmartIrrigation, but regarding the choice between (tmin+tmax)/2, it may not reflect the real average temperature of the day, as you may have longer hot periods or cold periods.
Today the temperature was quite balanced, computing the average between all the measurement samples led to 23.09°C, while the (min+max)/2 gave 23.5°C, but sometimes there may be a larger difference.

I think it would be important to know what the day temperature of OWM represents exactly : sliding average, or median, maybe ?

**Comment by karwosts** [comment]
I notice that OWM daily API provides 4 temperatures: day, morning, evening, and night. Possibly day temperature is the average of all the hours that they define as "daytime", but they don't really define it anywhere in their API doc. 

I will say that the actual research paper that these equations were based on use (Tmin+Tmax)/2. They don't actually call for the "average" temperature, and they may not be calibrated to use the average temperature. 

And yes I am personally OK to close this. I don't know enough about Github if this is my responsiblity to close, or yours, so go ahead and close if you want, or let me know and I'll try to do it.

**Comment by jeroenterheerdt** [comment]
yeah, as @karwosts says the research paper actually does `(min+max)/2` so that's what we adopted. Seems to work out fine.
